kernel/src/hil/screen.rs

```diff
@@ -0,0 +1,168 @@
+//! Interface for screens and displays.
+use crate::returncode::ReturnCode;
+
+pub enum ScreenRotation {
+    Normal,
+    Rotated90,
+    Rotated180,
+    Rotated270,
+}
+
+#[derive(Copy, Clone, PartialEq)]
+#[repr(usize)]
+#[allow(non_camel_case_types)]
+pub enum ScreenPixelFormat {
+    /// Pixels encoded as 1-bit, used for monochromatic displays
+    Mono,
+    /// Pixels encoded as 2-bit red channel, 3-bit green channel, 3-bit blue channel.
+    RGB_233,
+    /// Pixels encoded as 5-bit red channel, 6-bit green channel, 5-bit blue channel.
+    RGB_565,
+    /// Pixels encoded as 8-bit red channel, 8-bit green channel, 8-bit blue channel.
+    RGB_888,
+    /// Pixels encoded as 8-bit alpha channel, 8-bit red channel, 8-bit green channel, 8-bit blue channel.
+    ARGB_8888,
+    // other pixel formats may be defined.
+}
+
+impl ScreenPixelFormat {
+    pub fn get_bits_per_pixel(&self) -> usize {
+        match self {
+            Self::Mono => 1,
+            Self::RGB_233 => 8,
+            Self::RGB_565 => 16,
+            Self::RGB_888 => 24,
+            Self::ARGB_8888 => 32,
+        }
+    }
+}
+
+pub trait ScreenSetup {
+    fn set_client(&self, client: Option<&'static dyn ScreenSetupClient>);
+
+    /// Sets the screen resolution (in pixels). Returns ENOSUPPORT if the resolution is
+    /// not supported. The function should return SUCCESS if the request is registered
+    /// and will be sent to the screen.
+    /// Upon SUCCESS, the caller has to wait for the `command_complete` callback function
+    /// that will return the actual ReturnCode after setting the resolution.
+    fn set_resolution(&self, resolution: (usize, usize)) -> ReturnCode;
+
+    /// Sets the pixel format. Returns ENOSUPPORT if the pixel format is
+    /// not supported. The function should return SUCCESS if the request is registered
+    /// and will be sent to the screen.
+    /// Upon SUCCESS, the caller has to wait for the `command_complete` callback function
+    /// that will return the actual ReturnCode after setting the pixel format.
+    fn set_pixel_format(&self, depth: ScreenPixelFormat) -> ReturnCode;
+
+    /// Sets the rotation of the display.
+    /// The function should return SUCCESS if the request is registered
+    /// and will be sent to the screen.
+    /// Upon SUCCESS, the caller has to wait for the `command_complete` callback function
+    /// that will return the actual ReturnCode after setting the rotation.
+    ///
+    /// note this can swap the width with height.
```

Comment:
```suggestion
    /// Note that in the case of `Rotated90` or `Rotated270`, this will swap the width and height.
```

Can/should this say *will* like this? Is it a guarantee? Or a thing that some screen might do / expect?